.github/workflows/cleanup-azure-resources.yml

```diff
@@ -1,37 +1,104 @@
+name: CleanupAzureResources
+
 on:
   pull_request:
     types: [closed]
 
-name: CleanupAzureResources
+permissions:
+   id-token: write
+   contents: write
 
 jobs:
   cleanup:
-    runs-on: ubuntu-latest
+    runs-on: windows-latest # AzureAD PS module requires Windows.
+
     steps:
+      - name: Checkout Current Branch to access Custom Powershell Modules
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: "${{ github.base_ref }}"
+
+      # refer: https://microsofthealth.visualstudio.com/Health/_wiki/wikis/Health.wiki/163/OSS-CI
       - name: Login via Az module
         uses: azure/login@v1.4.1
         with:
-          creds: ${{ secrets.AZURE_CREDENTIALS }}
+          client-id: "${{secrets.AZ_DEVOPS_APP_ID}}"
+          tenant-id: "${{secrets.AZ_DEVOPS_TENANT_ID}}"
+          subscription-id: "${{secrets.AZ_DEVOPS_SUBSCRIPTION_ID}}"
           enable-AzPSSession: true
-      
-      - name: Clean Up Azure Resources
+
+      - name: Clean Up Azure Resource Groups
         uses: azure/powershell@v1.1.0
         with:
-          azPSVersion: '3.1.0'
+          azPSVersion: latest
           inlineScript: | 
-            [string]$PRNumber = "${{github.event.pull_request.number}}"
-            $groups = Get-AzResourceGroup -Name "dcm-pr$PRNumber-*-*"  
-            if ($groups -eq $null -or $groups.Count -eq 0)
-            {
-              Write-Host "No Resource groups related to PR $PRNumber are found."
+            $azContext = Get-AzContext
+            [string]$prNumber = "${{github.event.pull_request.number}}"
+            [string]$deploymentName = 'msh-dicom-pr-$(prNumber)'
+
+            Get-AzResourceGroup -DefaultProfile $azContext -Name $deploymentName | Remove-AzResourceGroup -Verbose -Force
+
+      - name: Purge key vault
+        uses: azure/powershell@v1.1.0
+        with:
+          azPSVersion: latest
+          inlineScript: | 
+            $azContext = Get-AzContext
+
+            [string]$resourceGroupRegion = 'southcentralus'
+            [string]$prNumber = "${{github.event.pull_request.number}}"
+            [string]$deploymentName = 'msh-dicom-pr-$(prNumber)'
```

Comment:
Do we have to calculate at each step? In the last step, we directly use the $(prNumber) variable, in other steps we calculate it